File: scripts/sync_docs_to_web.py
import shutil
from pathlib import Path


def main():
    root_dir = Path(__file__).parent.parent
    docs_dir = root_dir / "docs"
    readme_file = root_dir / "README.md"
    target_dir = root_dir / "apps" / "web" / "public" / "docs"

    if target_dir.exists():
        shutil.rmtree(target_dir)

    target_dir.mkdir(parents=True, exist_ok=True)

    if docs_dir.exists():
        for item in docs_dir.iterdir():
            if item.is_dir():
                shutil.copytree(item, target_dir / item.name, dirs_exist_ok=True)
            else:
                shutil.copy2(item, target_dir / item.name)
        print(f"Synced {docs_dir} to {target_dir}")
    else:
        print(f"Docs directory not found at {docs_dir}")

    if readme_file.exists():
        shutil.copy2(readme_file, target_dir / "ROOT-README.md")
        print(f"Synced {readme_file} to {target_dir / 'ROOT-README.md'}")
    else:
        print(f"Root README.md not found at {readme_file}")


if __name__ == "__main__":
    main()

File: scripts/sync_docs_to_web.js
#!/usr/bin/env node
// Syncs the /docs folder and root README.md into apps/web/public/docs
// so they are available as static assets in the web app.
// This replaces sync_docs_to_web.py so that no Python installation is required
// (e.g. inside the Node-based Docker builder image).

const { cpSync, mkdirSync, copyFileSync, existsSync, rmSync } = require("fs");
const { resolve } = require("path");

const rootDir = resolve(__dirname, "..");
const docsDir = resolve(rootDir, "docs");
const readmeFile = resolve(rootDir, "README.md");
const targetDir = resolve(rootDir, "apps", "web", "public", "docs");

rmSync(targetDir, { recursive: true, force: true });
mkdirSync(targetDir, { recursive: true });

if (existsSync(docsDir)) {
  cpSync(docsDir, targetDir, { recursive: true });
  console.log(`Synced ${docsDir} to ${targetDir}`);
} else {
  console.log(`Docs directory not found at ${docsDir}`);
}

if (existsSync(readmeFile)) {
  copyFileSync(readmeFile, resolve(targetDir, "ROOT-README.md"));
  console.log(`Synced ${readmeFile} to ${resolve(targetDir, "ROOT-README.md")}`);
} else {
  console.log(`Root README.md not found at ${readmeFile}`);
}
